### PR TITLE
VZ-6211: Yaml based Jaeger (from 1.3) should be upgradable

### PR DESCRIPTION
**Pull request by Abhijithang**
Jaeger component which is yaml based installation for 1.3.1, so this PR will enable jaeger to upgrade through helm to consequent release versions.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go [hunk 1]
```diff
@@ -397,6 +421,203 @@ func isCreateJaegerInstance(ctx spi.ComponentContext) (bool, error) {
...
+		return ctx.Log().ErrorfNewErr("Failed to get mutatingwebhookconfiguration %s: %v", ComponentMutatingWebhookConfigName, err)
+	}
+	err = kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Delete(context.TODO(), ComponentMutatingWebhookConfigName, metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return ctx.Log().ErrorfNewErr("Failed to delete mutatingwebhookconfiguration %s: %v", ComponentMutatingWebhookConfigName, err)
+	}
+	return nil
+}
+
+func removeValidatingWebhookConfig(ctx spi.ComponentContext) error {
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:500]: Function comment would be nice.

File: platform-operator/controllers/verrazzano/upgrade_component.go [hunk 2]
```diff
@@ -78,6 +78,7 @@ func (r *Reconciler) upgradeSingleComponent(spiCtx spi.ComponentContext, upgrade
 		case compStateInit:
 			// Check if component is installed, if not continue
 			installed, err := comp.IsInstalled(compContext)
+			compLog.Oncef("Now Checking for Component %s", compName)
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/upgrade_component.go:81]: Can this be made more specific.  What is is being checked?  I don't think component is capitalized in similar log entries.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator_component.go [hunk 3]
```diff
@@ -127,6 +147,56 @@ func (c jaegerOperatorComponent) ValidateUpdate(old *vzapi.Verrazzano, new *vzap
...
+	if err := ensureVerrazzanoMonitoringNamespace(ctx); err != nil {
+		return err
+	}
+	//_, err := DoManuallyCreatedJaegerInstanceExists(ctx)
+	installed, err := helmcli.IsReleaseInstalled(ComponentName, ComponentNamespace)
+	if err != nil {
+		ctx.Log().ErrorfNewErr("Failed searching for Jaeger release: %v", err)
+	}
+	if !installed && doDefaultJaegerInstanceDeploymentsExists(ctx) {
+		return ctx.Log().ErrorfNewErr("The jaeger resource %s/%s need to be removed or moved to a different namespace before continuing the upgrade, After the upgrade by default an Jaeger instance will be created in %s namespace: %v", ComponentNamespace, JaegerInstanceName, ComponentNamespace, err)
```
Comment (ayaranga) [inline platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator_component.go:163]: "Conflicting Jaeger instance %s/%s exists! Either disable the Verrazzano's default Jaeger instance creation by overriding jaeger.create Helm value for Jaeger Operator component to false or delete and recreate the existing Jaeger deployment in a different namespace." @kminder can suggest a better wording.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go (hunk 1 above)
Comment (Abhijithang) [reply platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:506]: Sorry, updated the comments.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator_component.go [hunk 4]
```diff
@@ -127,6 +147,56 @@ func (c jaegerOperatorComponent) ValidateUpdate(old *vzapi.Verrazzano, new *vzap
...
 }
 
+// PreUpgrade Jaeger component pre-upgrade processing
+func (c jaegerOperatorComponent) PreUpgrade(ctx spi.ComponentContext) error {
+	ctx.Log().Debugf("Jaeger pre-upgrade")
+	// Create the verrazzano-monitoring namespace if not already created
+	if err := ensureVerrazzanoMonitoringNamespace(ctx); err != nil {
+		return err
+	}
+	//_, err := DoManuallyCreatedJaegerInstanceExists(ctx)
```
[the remark this answers is not in the document]
Comment (Abhijithang) [reply platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator_component.go]: Done.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go [hunk 5]
```diff
@@ -295,13 +300,32 @@ func ensureVerrazzanoMonitoringNamespace(ctx spi.ComponentContext) error {
...
 		return ctx.Log().ErrorfNewErr("Failed to create or update the %s namespace: %v", ComponentNamespace, err)
 	}
 	return nil
 }
 
+// MutateVerrazzanoMonitoringNamespace modifies the given namespace for the Monitoring subcomponents
+// with the appropriate labels, in one location. If the provided namespace is not the Verrazzano
+// monitoring namespace, it is ignored.
+func MutateVerrazzanoMonitoringNamespace(ctx spi.ComponentContext, namespace *corev1.Namespace) {
+	if namespace.Name != constants.VerrazzanoMonitoringNamespace {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:315]: this name check is not needed but it doesn't hurt

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go [hunk 6]
```diff
@@ -295,13 +300,32 @@ func ensureVerrazzanoMonitoringNamespace(ctx spi.ComponentContext) error {
...
+		return
+	}
+	if namespace.Labels == nil {
+		namespace.Labels = map[string]string{}
+	}
+	namespace.Labels[globalconst.LabelVerrazzanoNamespace] = constants.VerrazzanoMonitoringNamespace
+
+	istio := ctx.EffectiveCR().Spec.Components.Istio
+	if istio != nil && istio.IsInjectionEnabled() {
+		namespace.Labels[globalconst.LabelIstioInjection] = "enabled"
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:325]: Aren't there other components in the namespace, like Prometheus?

**Review by kminder — APPROVED** [review]
Approved.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go (hunk 6 above)
Comment (Abhijithang) [reply platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:325]: Also, I guess PrometheusOperator was disabled as part of 1.3.1 release.

**Pull request merged**